core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -440,20 +420,21 @@ public boolean entryExpiredInMemoryFromIteration(InternalCacheEntry<K, V> entry,
             // the entire value.  Unfortunately this could cause a concurrent write to be undone
             resultStage = cacheToUse.removeLifespanExpired(key, null, null);
          }
-         return CompletionStages.ignoreValue(resultStage.whenComplete((b, t) -> {
+         return resultStage.handle((b, t) -> {
             expiring.remove(key);
             if (t != null) {
                completableFuture.completeExceptionally(t);
             } else {
-               completableFuture.complete(b);
+               completableFuture.complete(null);
```

Comment:
I think I might have set it to null to make sure that the consumers of the future can deal with null more deterministically (instead of waiting for a random test failure), but I reverted the change.

Going back here made me remove some usages of `CompletionStages.ignoreValue()`, but then I also realized that `ignoreValue()` doesn't really need to allocate a new `CompletableFuture`: it can use type erasure instead to pretend that the input stage was a `CompletionStage<Void>` all along.